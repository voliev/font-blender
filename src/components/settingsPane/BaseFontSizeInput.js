import React from "react";
import debounce from "lodash.debounce";
import PropTypes from "prop-types";
import "./styles/BaseFontSizeInput.css";

const BaseFontSizeInput = ({ disabled, value, fontSizeChangeHandler }) => (
  <div className="BaseFontSizeInput">
    <label className="label" htmlFor="base-font-size">
      Font size (px)
    </label>
    <input
      disabled={disabled}
      id="base-font-size"
      min="1"
      max="50"
      name="fontSize"
      onChange={debounce(fontSizeChangeHandler, 100, {
        leading: true
      })}
      type="number"
      value={value}
    />
  </div>
);

BaseFontSizeInput.propTypes = {
  disabled: PropTypes.bool.isRequired,
  value: PropTypes.number.isRequired,
  fontSizeChangeHandler: PropTypes.func.isRequired
};

export default React.memo(BaseFontSizeInput);
