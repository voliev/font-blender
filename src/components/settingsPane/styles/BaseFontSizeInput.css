.BaseFontSizeInput {
  display: flex;
  flex-direction: column;
  margin: 0 0 var(--space-xs) 0;
}
