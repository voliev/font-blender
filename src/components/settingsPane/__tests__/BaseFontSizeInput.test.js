import React from "react";
import { mount, shallow } from "enzyme";
import BaseFontSizeInput from "../BaseFontSizeInput";

const setup = (renderFn = shallow, propsOverrides = {}) => {
  const props = {
    disabled: false,
    fontSizeChangeHandler: jest.fn(),
    value: 16,
    ...propsOverrides
  };

  return {
    ...props,
    wrapper: renderFn(<BaseFontSizeInput {...props} />)
  };
};

afterEach(() => {
  jest.clearAllMocks();
});

it("should render without crashing", () => {
  const { wrapper } = setup();

  expect(wrapper).toMatchSnapshot();
});

it("should receive corresponding props", () => {
  const { disabled, fontSizeChangeHandler, value, wrapper } = setup(mount);

  expect(typeof wrapper.prop("value")).toBe("number");
  expect(wrapper.prop("value")).toBe(value);

  expect(typeof wrapper.prop("disabled")).toBe("boolean");
  expect(wrapper.prop("disabled")).toBe(disabled);

  expect(typeof wrapper.prop("fontSizeChangeHandler")).toBe("function");
  expect(wrapper.prop("fontSizeChangeHandler")).toBe(fontSizeChangeHandler);
});

it("should handle input change", () => {
  const { fontSizeChangeHandler, wrapper } = setup();

  const event = {
    target: {
      value: 18
    }
  };

  wrapper.find("input").simulate("change", event);

  expect(fontSizeChangeHandler).toHaveBeenCalledTimes(1);
  expect(fontSizeChangeHandler).toHaveBeenCalledWith(event);
});

it("should debounce handler on input change", () => {
  const { fontSizeChangeHandler, wrapper } = setup();

  const input = wrapper.find("input");

  for (let i = 0; i < 100; i++) {
    input.simulate("change");
  }

  expect(fontSizeChangeHandler).toHaveBeenCalledTimes(1);
});
