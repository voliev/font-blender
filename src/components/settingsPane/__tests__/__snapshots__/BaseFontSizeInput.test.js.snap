// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without crashing 1`] = `
<div
  className="BaseFontSizeInput"
>
  <label
    className="label"
    htmlFor="base-font-size"
  >
    Font size (px)
  </label>
  <input
    disabled={false}
    id="base-font-size"
    max="50"
    min="1"
    name="fontSize"
    onChange={[Function]}
    type="number"
    value={16}
  />
</div>
`;
